File: src/tests/fidl/source_compatibility/enum-member-remove/rust/step_00_before.rs
// Copyright 2021 The Fuchsia Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

#![allow(dead_code)]

use fidl_fidl_test_enummemberremove as fidl_lib;

// [START contents]
fn writer(s: &str) -> fidl_lib::Color {
    match s {
        "red" => fidl_lib::Color::Red,
        "blue" => fidl_lib::Color::Blue,
        "yellow" => fidl_lib::Color::Yellow,
        _ => fidl_lib::Color::unknown(),
    }
}

fn reader(color: fidl_lib::Color) -> &'static str {
    match color {
        fidl_lib::Color::Red => "red",
        fidl_lib::Color::Blue => "blue",
        fidl_lib::Color::Yellow => "yellow",
        fidl_lib::ColorUnknown!() => "unknown",
    }
}
// [END contents]

fn main() {}

File: src/tests/fidl/source_compatibility/enum-strict-flexible/llcpp/step_00_before.cc
// Copyright 2021 The Fuchsia Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

#include <fidl/fidl.test.enumstrictflexible/cpp/wire.h>  // nogncheck
namespace fidl_test = fidl_test_enumstrictflexible;

// [START contents]
template <fidl_test::wire::Color color>
class ComplementaryColors {};

fidl_test::wire::Color writer(std::string s) {
  if (s == "red") {
    return fidl_test::wire::Color::kRed;
  } else if (s == "blue") {
    return fidl_test::wire::Color::kBlue;
  } else {
    return fidl_test::wire::Color::kUnknownColor;
  }
}

std::string reader(fidl_test::wire::Color color) {
  switch (color) {
    case fidl_test::wire::Color::kRed:
      return "red";
    case fidl_test::wire::Color::kBlue:
      return "blue";
    case fidl_test::wire::Color::kUnknownColor:
      return "unknown";
    default:
      return "error";
  }
}
// [END contents]

int main(int argc, const char** argv) { return 0; }
